parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -414,4 +414,16 @@ write_files:
   encoding: gzip
   owner: root
   content: !!binary |
-    {{GetVariableProperty "cloudInitData" "customSearchDomainsScript"}}
\ No newline at end of file
+    {{GetVariableProperty "cloudInitData" "customSearchDomainsScript"}}
+
+{{ if EnableBootstrapProfileContainerRegistry -}}
+# TODO(binxi): need to update for sovereign cloud.
+- path: /etc/containerd/certs.d/mcr.microsoft.com/hosts.toml
+  permissions: "0644"
+  owner: root
+  content: |
+    server = "https://mcr.microsoft.com"
+
+    [host."https://{{GetBootstrapProfileContainerRegistryServer}}"]
+      capabilities = ["pull", "resolve"]
+{{- end}}
```

Comment:
>}} [](http://example.com/codeflow?start=7&length=2)

some whitespace changes are not necessary in testdata

can you tweak {{- --}} to remove unneed whitespace when the feature is off?

Reply:
removed unneeded whitespace when the feature is off